File: SDKs/digitalocean.com/2.0/go/pkg/models/operations/deletefirewalldroplets.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DeleteFirewallDropletsPathParams struct {
	FirewallID string `pathParam:"style=simple,explode=false,name=firewall_id"`
}

type DeleteFirewallDropletsRequest struct {
	PathParams DeleteFirewallDropletsPathParams
	Request    *interface{} `request:"mediaType=application/json"`
}

type DeleteFirewallDroplets401ApplicationJSON struct {
	ID        string  `json:"id"`
	Message   string  `json:"message"`
	RequestID *string `json:"request_id"`
}

type DeleteFirewallDropletsResponse struct {
	ContentType                                               string
	Headers                                                   map[string][]string
	StatusCode                                                int64
	DeleteFirewallDroplets401ApplicationJSONObject            *DeleteFirewallDroplets401ApplicationJSON
	Onev211ClicksGetResponses401ContentApplication1jsonSchema *shared.Onev211ClicksGetResponses401ContentApplication1jsonSchema
}

File: SDKs/digitalocean.com/2.0/typescript/src/sdk/models/operations/deletefirewalldroplets.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class DeleteFirewallDropletsPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=firewall_id" })
  firewallId: string;
}


export class DeleteFirewallDropletsRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: DeleteFirewallDropletsPathParams;

  @Metadata({ data: "request, media_type=application/json" })
  request?: any;
}


export class DeleteFirewallDroplets401ApplicationJson extends SpeakeasyBase {
  @Metadata({ data: "json, name=id" })
  id: string;

  @Metadata({ data: "json, name=message" })
  message: string;

  @Metadata({ data: "json, name=request_id" })
  requestId?: string;
}


export class DeleteFirewallDropletsResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  headers: Map<string, string[]>;

  @Metadata()
  statusCode: number;

  @Metadata()
  deleteFirewallDroplets401ApplicationJsonObject?: DeleteFirewallDroplets401ApplicationJson;

  @Metadata()
  onev211ClicksGetResponses401ContentApplication1jsonSchema?: shared.Onev211ClicksGetResponses401ContentApplication1jsonSchema;
}
